src/qililab/typings/experiment.py

```diff
@@ -35,7 +32,7 @@ class ExperimentSettings:
 
     def __str__(self):
         """Returns a string representation of the experiment settings."""
-        return str(YAML().dump(asdict(self), io.BytesIO()))
```

Comment:
Why this change? 
The previous seems more general, and more adaptable to future changes 🤔

Reply:
I made it more general using dictionaries like before but using `asdict` directly as imo is a more readable code. But can go back to it if you dont think the same.

Reply:
Maybe leave the test as it used to be, to see if we get the same string as before, or not, no?

Reply:
I think you can definelty change the implementation, and make it better, and how you made it now looks amazing, but at the same time, I think we shouldn't change the final string representation that cames out of it, respect how it was, for backwards compatibility, dunno if it is used somewhere this :)

Reply:
What I meant, in a more clear way, is:

Does`str(YAML().dump(asdict(self), io.BytesIO()))` == `str(asdict(self))`?

Reply:
Not actually, I have tested the preious code and I get `'None'` whereas with the new code we get `"{'hardware_average': 1024, 'software_average': 1, 'repetition_duration': 200000, 'num_bins': 1}"`

Reply:
Cool then! 🙌  Resolving this!